### DISCUSSION TITLE
Real time plots for PnP and other optimizaition algorithms

### DISCUSSION THREAD

**Discussion by David-Morales-Norato** [deepinv/deepinv#74] [Ideas]
Hello,

I wanted to express my gratitude for providing this incredible repository. I am currently working on a paper that involves the use of Pnp, and I am really excited to utilize your code for it. However, I have a specific need regarding real-time monitoring of the reconstructions to analyze the algorithm's performance, ensuring that the loss is decreasing as expected, and so on. I noticed that your code generates the report after the reconstruction process, but for time-consuming reconstructions, this approach becomes impractical.

I have my own implementation of PnP that includes real-time monitoring, and I have attached some screen captures from Tensorboard to illustrate this feature.

I strongly believe that integrating this real-time monitoring feature would be immensely beneficial not only for PnP but also for the other algorithms you have implemented. While I thought about implementing this feature myself and submitting a pull request, I encountered difficulties due to the extensive and complex code in the BaseOptim class. It has been challenging for me to understand most of it, which makes it hard to know where to start.

I sincerely hope you will consider adding this real-time monitoring feature to your project. Alternatively, if you could provide me with some guidance or ideas on how to implement it myself, I would be happy to do it.

Thank you once again for your remarkable work.

Best regards,

![image](https://github.com/deepinv/deepinv/assets/46037413/dfadea0b-1b3f-427d-a4e7-747c33374a5c)
![image](https://github.com/deepinv/deepinv/assets/46037413/2dde3a2d-4893-417c-8008-09b8b62b2c0e)

**Comment by tachella** [comment]
Hi David! 

Sorry for the very late reply - we had completely missed your post at the time of posting.

I think the idea is quite nice, maybe you could open a PR with a minimal working example if this is still currently interesting?
